### DISCUSSION TITLE
How to get gen2 function working?

### DISCUSSION THREAD

**Discussion by thecannabisapp** [firebase/functions-samples#1038] [Q&A] [answered]
I've followed the docs, but I get a 500 error when I call the function.

<img width="1007" alt="Screenshot 2023-02-06 at 23 10 18" src="https://user-images.githubusercontent.com/58605231/217108100-a1109658-6295-4836-b2a9-8024106c57d3.png">

Below is my code from the client.

```typescript
      try {
        const response = await httpsCallableFromURL<unknown, IToken>(
          functions,
          "https://getstreamtokenv2-nnvm3z4kea-ez.a.run.app/getstreamtokenv2"
        )();
        return response.data;
      } catch (error) {
        throw new Error("token error");
      }
```

**Comment by ryanmio** [comment, marked as answer]
Your 500 error indicates a server-side issue. The client code is simply the caller and can't fix a 500 error coming from the server. Make sure the function at `https://getstreamtokenv2-nnvm3z4kea-ez.a.run.app/getstreamtokenv2` is working as expected.

### Debugging Steps:

1. **Check Server Logs**: Look at the server-side logs to identify the cause of the 500 error.
  
2. **Local Testing**: If possible, test the function locally before deploying.

3. **Dependencies**: Ensure all dependencies are correctly installed on the server.

4. **Request and Response**: Ensure that the request and response formats are correct according to your function's expectations.

Here's an example of how you might log errors in your cloud function:
```javascript
exports.yourFunction = functions.https.onCall((data, context) => {
  try {
    // Your code here
  } catch (error) {
    console.error(error);
    throw new functions.https.HttpsError('internal', 'An internal error occurred');
  }
});
```

This should give you a clearer picture of what might be going wrong on the server-side.
